docs/rest_v2.md

```diff
@@ -0,0 +1,821 @@
+
+#### type Client
+
+```go
+type Client struct {
+
+	// service providers
+	Candles     CandleService
+	Orders      OrderService
+	Positions   PositionService
+	Trades      TradeService
+	Tickers     TickerService
+	Currencies  CurrenciesService
+	Platform    PlatformService
+	Book        BookService
+	Wallet      WalletService
+	Ledgers     LedgerService
+	Stats       StatsService
+	Status      StatusService
+	Derivatives DerivativesService
+	Funding     FundingService
+
+	Synchronous
+}
+```
+
+
+#### func  NewClient
+
+```go
+func NewClient() *Client
+```
+Create a new Rest client
+
+#### func  NewClientWithHttpDo
+
+```go
+func NewClientWithHttpDo(httpDo func(c *http.Client, r *http.Request) (*http.Response, error)) *Client
+```
+Create a new Rest client with a custom http handler
+
+#### func  NewClientWithSynchronousNonce
+
+```go
+func NewClientWithSynchronousNonce(sync Synchronous, nonce utils.NonceGenerator) *Client
+```
+Create a new Rest client with a synchronous HTTP handler and a custom nonce
+generaotr
+
+#### func  NewClientWithSynchronousURLNonce
+
+```go
+func NewClientWithSynchronousURLNonce(sync Synchronous, url string, nonce utils.NonceGenerator) *Client
+```
+Create a new Rest client with a synchronous HTTP handler and a custom base url
+and nonce generator
+
+#### func  NewClientWithURL
+
+```go
+func NewClientWithURL(url string) *Client
+```
+Create a new Rest client with a custom base url
+
+#### func  NewClientWithURLHttpDo
+
+```go
+func NewClientWithURLHttpDo(base string, httpDo func(c *http.Client, r *http.Request) (*http.Response, error)) *Client
+```
+Create a new Rest client with a custom base url and HTTP handler
+
+#### func  NewClientWithURLHttpDoNonce
+
+```go
+func NewClientWithURLHttpDoNonce(base string, httpDo func(c *http.Client, r *http.Request) (*http.Response, error), nonce utils.NonceGenerator) *Client
+```
+Create a new Rest client with a custom base url, HTTP handler and none generator
+
+#### func  NewClientWithURLNonce
+
+```go
+func NewClientWithURLNonce(url string, nonce utils.NonceGenerator) *Client
+```
+Create a new Rest client with a custom nonce generator
+
+#### func (*Client) Credentials
+
+```go
+func (c *Client) Credentials(key string, secret string) *Client
+```
+Set the clients credentials in order to make authenticated requests
+
+#### func (*Client) NewAuthenticatedRequest
+
+```go
+func (c *Client) NewAuthenticatedRequest(permissionType bitfinex.PermissionType, refURL string) (Request, error)
+```
+Create a new authenticated GET request with the given permission type and
+endpoint url For example permissionType = "r" and refUrl = "/orders" then the
+target endpoint will be https://api.bitfinex.com/v2/auth/r/orders/:Symbol
+
+#### func (*Client) NewAuthenticatedRequestWithBytes
+
+```go
+func (c *Client) NewAuthenticatedRequestWithBytes(permissionType bitfinex.PermissionType, refURL string, data []byte) (Request, error)
```

Comment:
I leave it up to you, but you may want to consider making this and a few other functions private, excluding them from the docs.